### PR TITLE
feat(admin-ui): apollo updated to v3

### PR DESCRIPTION
**Pull request by mgs95**

### CODE REVIEW

**Comment by mgs95** [comment]
Closes #240

File: admin/admin-ui/src/App.tsx [hunk 1]
```diff
@@ -109,19 +112,11 @@ export function Routes() {
...
-  function closeLogs() {
-    client.writeData({ data: { logsOpened: false } });
-  }
+  const { openLogs, closeLogs } = useLogs();
 
   const handlers = {
-    CLOSE_LOGS: closeLogs,
-    OPEN_LOGS: openLogs
+    CLOSE_LOGS: openLogs,
+    OPEN_LOGS: closeLogs
```
Comment (angelsanchez) [inline admin/admin-ui/src/App.tsx:119]: CLOSE_LOGS: openLogs

OPEN_LOGS: closeLogs

is this correct?

Comment (mgs95) [reply admin/admin-ui/src/App.tsx]: It was wrong. Fixed

File: admin/admin-ui/src/Components/NotificationService/Services/RuntimeCreated/RuntimeCreated.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      next(data) {
+        const runtime: runtimeCreated_watchRuntimeCreated = get(
+          data,
+          'data.watchRuntimeCreated'
+        );
+        const runtimePath = ROUTE.RUNTIME.replace(':runtimeId', runtime.id);
+
+        const newNotification: Notification = {
+          id: `runtime-${runtime.id}-created`,
+          message: `The RUNTIME "${runtime.name}" has been successfully created!`,
```
[the remark this answers is not in the document]
Comment (mgs95) [reply admin/admin-ui/src/Components/NotificationService/Services/RuntimeCreated/RuntimeCreated.tsx:38]: Is the way is shown in the designs:
https://projects.invisionapp.com/d/main#/console/18748947/401758486/preview

File: admin/admin-ui/src/Graphql/client/cache.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        logs: { read: () => logs() },
+        notifications: { read: () => notifications() },
+        logTabs: { read: () => logTabs() },
+        activeTabId: { read: () => activeTabId() },
+        logsOpened: { read: () => logsOpened() },
+        openedVersion: { read: () => openedVersion() },
+        userSettings: { read: () => userSettings() },
+        runtimes: { merge: false },
+        versions: { merge: false },
+        settings: { merge: false }
```
[the remark this answers is not in the document]
Comment (mgs95) [reply admin/admin-ui/src/Graphql/client/cache.tsx:54]: merge function follows the following format: ```merge(existing = [], incoming: any[]): void```.
`merge: true` only works for objects as is a shortcut from `options.mergeObjects(existing, incoming)`.

Our API returns always the complete set of elements so there is not a reason for merging old values with new ones (merging will require to check for repeated values).

**Review by angelsanchez — APPROVED** [review]

**Pull request merged**